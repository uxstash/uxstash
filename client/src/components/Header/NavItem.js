import React from "react";
import { NavLink, withRouter } from "react-router-dom";

export const NavItem = props => {
  return (
    <li className={props.to === props.location.pathname ? "active" : ""}>
      <NavLink to={props.to}>
        {props.title}
        <small>{props.description}</small>
      </NavLink>
    </li>
  );
};

export default withRouter(NavItem);
