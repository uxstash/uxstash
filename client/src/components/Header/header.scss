@import '../../styles/sass/variables';

.main-header {
  nav ul {
    list-style: none;
    display: flex;

    li {
      flex: 1;
      margin-left: 4rem;
      margin-top: .5rem;
      border-bottom: 2px solid transparent;

      &:not(.brand) {

        &:hover,
        &:focus,
        &.active {
          border-bottom-color: $off-white;
        }
      }

      &.brand {
        margin-top: 0;
        margin-left: 0;
      }

      a {
        display: inline-block;
        padding: 1rem 0;
      }
    }
  }

  .logo {
    display: flex;
    align-items: center;

    img {
      margin-right: 1rem;
    }
  }

}