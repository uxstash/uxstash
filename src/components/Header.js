import React from 'react';
import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Nav from './Nav';
import logo from '../images/ux_stash.svg';

const MainHeader = styled.header`
  display: grid;
  @media (min-width: ${props => props.theme.md}) {
    grid-template-columns: repeat(6, 1fr);
  }
  grid-column-gap: 20px;
  margin-bottom: 94px;
  padding-top: 45px;

  .site-logo {
    @media (min-width: ${props => props.theme.md}) {
      grid-area: 1 / 1 / 2 / 2;
    }
    display: flex;
    align-items: center;

    > a {
      font-weight: bold;
    }
  }

  nav {
    grid-area: 1 / 2 / 2 / 3;
    @media (min-width: ${props => props.theme.md}) {
      grid-area: 1 / 2 / 2 / 7;
    }
  }

  object {
    margin-right: 16px;
  }

  a {
    color: ${props => props.theme.offWhite};
    text-decoration: none;
  }
`;

const Header = ({ siteTitle }) => (
  <MainHeader>
    <div className="site-logo">
      <object type="image/svg+xml" data={logo} />
      <Link to="/">{siteTitle}</Link>
    </div>
    <Nav />
  </MainHeader>
);

Header.propTypes = {
  siteTitle: PropTypes.string,
};

Header.defaultProps = {
  siteTitle: '',
};

export default Header;
