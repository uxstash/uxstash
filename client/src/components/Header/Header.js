import React from "react";
import { NavLink } from "react-router-dom";
import NavItem from "./NavItem";
import "./header.scss";
import categories from "../../data/categories";

class Header extends React.Component {
  render() {
    return (
      <header className="main-header">
        <div className="container">
          <nav>
            <ul>
              <li className="brand">
                <NavLink to="/" className="logo">
                  <img src="/ux_stash.svg" alt="UX Stash Logo" />
                  UX Stash
                </NavLink>
              </li>
              {categories.map((category, index) => {
                return (
                  <NavItem
                    title={category.title}
                    subtitle={category.subtitle}
                    to={`/${category.title.toLowerCase()}`}
                    key={index}
                  />
                );
              })}
            </ul>
          </nav>
        </div>
      </header>
    );
  }
}

export default Header;
