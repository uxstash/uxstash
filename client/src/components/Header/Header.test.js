import React from "react";
import { shallow, mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";

import Header from "./Header";

describe("Header", () => {
  it("should render correctly in debug mode", () => {
    const component = shallow(<Header debug />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });

  it("should render 6 list items", () => {
    const component = shallow(<Header />);

    expect(component.find("ul").children()).toHaveLength(6);
  });

  it("should only render 1 active link and 1 brand", () => {
    const component = mount(
      <MemoryRouter initialEntries={["/strategy"]}>
        <Route component={Header} />
      </MemoryRouter>
    );
    expect(component.find("li.brand")).toHaveLength(1);
    expect(component.find("li.active")).toHaveLength(1);
  });
});
