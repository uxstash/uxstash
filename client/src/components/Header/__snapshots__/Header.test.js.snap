// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly in debug mode 1`] = `
<header
  className="main-header"
>
  <div
    className="container"
  >
    <nav>
      <ul>
        <li
          className="brand"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="logo"
            to="/"
          >
            <img
              alt="UX Stash Logo"
              src="/ux_stash.svg"
            />
            UX Stash
          </NavLink>
        </li>
        <withRouter(NavItem)
          key="0"
          subtitle="Needs & Objectives"
          title="Strategy"
          to="/strategy"
        />
        <withRouter(NavItem)
          key="1"
          subtitle="Specs & Requirements"
          title="Management"
          to="/management"
        />
        <withRouter(NavItem)
          key="2"
          subtitle="Structure & Interaction"
          title="Architecture"
          to="/architecture"
        />
        <withRouter(NavItem)
          key="3"
          subtitle="Code & Navigation"
          title="Development"
          to="/development"
        />
        <withRouter(NavItem)
          key="4"
          subtitle="Visuals & Needs"
          title="Design"
          to="/design"
        />
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header should render correctly with no props 1`] = `
<header
  className="main-header"
>
  <div
    className="container"
  >
    <nav>
      <ul>
        <li
          className="brand"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="logo"
            to="/"
          >
            <img
              alt="UX Stash Logo"
              src="/ux_stash.svg"
            />
            UX Stash
          </NavLink>
        </li>
        <withRouter(NavItem)
          key="0"
          subtitle="Needs & Objectives"
          title="Strategy"
          to="/strategy"
        />
        <withRouter(NavItem)
          key="1"
          subtitle="Specs & Requirements"
          title="Management"
          to="/management"
        />
        <withRouter(NavItem)
          key="2"
          subtitle="Structure & Interaction"
          title="Architecture"
          to="/architecture"
        />
        <withRouter(NavItem)
          key="3"
          subtitle="Code & Navigation"
          title="Development"
          to="/development"
        />
        <withRouter(NavItem)
          key="4"
          subtitle="Visuals & Needs"
          title="Design"
          to="/design"
        />
      </ul>
    </nav>
  </div>
</header>
`;
